credentials/google/google.go

```diff
@@ -64,14 +64,25 @@ func NewDefaultCredentials() credentials.Bundle {
 //
 // This API is experimental.
 func NewComputeEngineCredentials() credentials.Bundle {
+	return NewComputeEngineCredsWithPerRPC(nil)
+}
+
+// NewComputeEngineCredsWithPerRPC returns a credentials bundle that is configured to work
+// with google services. This API must only be used when running on GCE.
+//
+// This API is experimental.
+func NewComputeEngineCredsWithPerRPC(perRPC credentials.PerRPCCredentials) credentials.Bundle {
 	c := &creds{
 		newPerRPCCreds: func() credentials.PerRPCCredentials {
+			if perRPC != nil {
```

Comment:
And as mentioned in another comment, please move this if check out of the closure (you can make two closures, based on the value of `perRPC`)

Reply:
Done